### ISSUE TITLE
Problem using assertj-core with OSGi test

### ISSUE THREAD

**Issue by cvgaviao** [joel-costigliola/assertj-core#701]
#### Summary
The default bndlib setup sometimes add optional dependencies in the import-package without need.
In case of asserj-core, bndlib is adding those packages below. Note that there are some ant and some internal packages:

> Import-Package: javax.xml.parsers,**org.apache.tools.ant,org.apache.tools.
 ant.types**,org.assertj.core.api;version="[3.4,4)",org.assertj.core.api.e
 xception;version="[3.4,4)",org.assertj.core.api.filter;version="[3.4,4)
 ",org.assertj.core.api.iterable;version="[3.4,4)",org.assertj.core.cond
 ition;version="[3.4,4)",org.assertj.core.data;version="[3.4,4)",org.ass
 ertj.core.description;version="[3.4,4)",org.assertj.core.error;version=
 "[3.4,4)",org.assertj.core.error.future;version="[3.4,4)",org.assertj.c
 ore.error.uri;version="[3.4,4)",org.assertj.core.extractor;version="[3.
 4,4)",org.assertj.core.groups;version="[3.4,4)",org.assertj.core.intern
 al;version="[3.4,4)",org.assertj.core.internal.cglib.asm,org.assertj.co
 re.internal.cglib.core,org.assertj.core.internal.cglib.proxy,org.assert
 j.core.internal.cglib.reflect,org.assertj.core.internal.cglib.transform
 ,org.assertj.core.presentation;version="[3.4,4)",org.assertj.core.util;
 version="[3.4,4)",org.assertj.core.util.diff;version="[3.4,4)",org.asse
 rtj.core.util.diff.myers;version="[3.4,4)",org.assertj.core.util.intros
 pection;version="[3.4,4)",org.assertj.core.util.xml;version="[3.4,4)",o
 rg.junit.rules,org.junit.runner,org.junit.runners.model,org.w3c.dom,org
 .w3c.dom.bootstrap,org.w3c.dom.ls,org.xml.sax

 so, the bundle-maven-plugin configuration must be modified in order to prevent internal packages to be exported and also prevent import ant packages (or turn it optional)

**Comment by cvgaviao** [comment]
I saw in the code that this issue was already fixed and released in 3.5.1 version.

**Comment by PascalSchumacher** [comment]
I guess this can be closed, now that the pull request has been merged?

**Comment by cvgaviao** [comment]
@PascalSchumacher , I am not seeing a commit in master branch,the PR was only accept for  2.x branch

**Comment by PascalSchumacher** [comment]
@joel-costigliola merges all commits from 2.x to master from time to time. So the next 3.x release will included your fix.

**Comment by joel-costigliola** [comment]
@cvgaviao are you sure this is gonna solve your problem ? I have generated the manifest again and the only difference that I see is the order of `version="2.5.1";` and `uses:=` clauses.

Have you tested the jar generated from your commit ?

**Comment by cvgaviao** [comment]
@joel-costigliola , are you saying that in the machine you are building the jar its manifest still have "Import-Package: javax.xml.parsers,org.apache.tools.ant,org.apache.tools.ant.types," ?

**Comment by joel-costigliola** [comment]
There are no such packages in 2.5.x, I'll have a look at 3.5.x

**Comment by joel-costigliola** [comment]
here's the manifest for 3.5.2-SNAPSHOT, `javax.xml.parsers` is still present.

Is it sufficient to solve your problem ? 
Can you test the latest master branch ? it contains your fix.

```
Manifest-Version: 1.0
Build-Jdk: 1.8.0_77
Built-By: joel
Bundle-Description: Rich and fluent assertions for testing for Java
Bundle-DocURL: joel-costigliola.github.io/assertj/index.html
Bundle-License: http://www.apache.org/licenses/LICENSE-2.0.txt
Bundle-ManifestVersion: 2
Bundle-Name: AssertJ fluent assertions
Bundle-RequiredExecutionEnvironment: JavaSE-1.8
Bundle-SymbolicName: org.assertj.core
Bundle-Vendor: AssertJ
Bundle-Version: 3.5.2.SNAPSHOT
Created-By: Apache Maven Bundle Plugin
Export-Package: org.assertj.core.api;uses:="net.sf.cglib.proxy,org.asser
 tj.core.api.filter,org.assertj.core.api.iterable,org.assertj.core.condi
 tion,org.assertj.core.data,org.assertj.core.description,org.assertj.cor
 e.error,org.assertj.core.groups,org.assertj.core.internal,org.assertj.c
 ore.presentation,org.junit.rules,org.junit.runner,org.junit.runners.mod
 el";version="3.5.2",org.assertj.core.api.exception;version="3.5.2",org.
 assertj.core.api.filter;uses:="org.assertj.core.api";version="3.5.2",or
 g.assertj.core.api.iterable;version="3.5.2",org.assertj.core.condition;
 uses:="org.assertj.core.api";version="3.5.2",org.assertj.core.data;vers
 ion="3.5.2",org.assertj.core.description;version="3.5.2",org.assertj.co
 re.error;uses:="org.assertj.core.api,org.assertj.core.data,org.assertj.
 core.description,org.assertj.core.internal,org.assertj.core.presentatio
 n,org.assertj.core.util.diff";version="3.5.2",org.assertj.core.error.fu
 ture;uses:="org.assertj.core.error";version="3.5.2",org.assertj.core.er
 ror.uri;uses:="org.assertj.core.error";version="3.5.2",org.assertj.core
 .extractor;uses:="org.assertj.core.api.iterable,org.assertj.core.groups
 ";version="3.5.2",org.assertj.core.groups;uses:="org.assertj.core.api.i
 terable";version="3.5.2",org.assertj.core.internal;uses:="org.assertj.c
 ore.api,org.assertj.core.data,org.assertj.core.error,org.assertj.core.u
 til.diff";version="3.5.2",org.assertj.core.presentation;uses:="org.asse
 rtj.core.data,org.assertj.core.groups";version="3.5.2",org.assertj.core
 .util;uses:="org.assertj.core.api.filter,org.assertj.core.presentation"
 ;version="3.5.2",org.assertj.core.util.diff;uses:="org.assertj.core.uti
 l.diff.myers";version="3.5.2",org.assertj.core.util.diff.myers;uses:="o
 rg.assertj.core.util.diff";version="3.5.2",org.assertj.core.util.intros
 pection;version="3.5.2",org.assertj.core.util.xml;version="3.5.2"
Import-Package: javax.xml.parsers,net.sf.cglib.proxy;resolution:=optiona
 l,org.assertj.core.api,org.assertj.core.api.exception,org.assertj.core.
 api.filter,org.assertj.core.api.iterable,org.assertj.core.condition,org
 .assertj.core.data,org.assertj.core.description,org.assertj.core.error,
 org.assertj.core.error.future,org.assertj.core.error.uri,org.assertj.co
 re.extractor,org.assertj.core.groups,org.assertj.core.internal,org.asse
 rtj.core.presentation,org.assertj.core.util,org.assertj.core.util.diff,
 org.assertj.core.util.diff.myers,org.assertj.core.util.introspection,or
 g.assertj.core.util.xml,org.junit.rules,org.junit.runner,org.junit.runn
 ers.model,org.w3c.dom,org.w3c.dom.bootstrap,org.w3c.dom.ls,org.xml.sax
Private-Package: org.assertj.core.internal,org.assertj.core.extractor,or
 g.assertj.core.condition,org.assertj.core.description,org.assertj.core.
 data,org.assertj.core.presentation,org.assertj.core.error,org.assertj.c
 ore.error.uri,org.assertj.core.error.future,org.assertj.core.groups,org
 .assertj.core.api,org.assertj.core.api.iterable,org.assertj.core.api.ex
 ception,org.assertj.core.api.filter,org.assertj.core.util.xml,org.asser
 tj.core.util,org.assertj.core.util.introspection,org.assertj.core.util.
 diff,org.assertj.core.util.diff.myers
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
Tool: Bnd-3.0.0.201509101326
```

**Comment by cvgaviao** [comment]
@joel-costigliola, I tested and its ok. the package _javax.xml.parsers_ was bold wrongly by me because it comes from jdk... the problem was the ant one.
thanks

**Comment by joel-costigliola** [comment]
thanks for the feedback.

**Comment by joel-costigliola** [comment]
I have just released assertj-core 3.5.2 with this fix.
